src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java

```diff
@@ -0,0 +1,170 @@
+package seedu.address.model.appointment;
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.appointment.exceptions.AppointmentNotFoundException;
+import seedu.address.model.appointment.exceptions.DuplicateAppointmentsException;
+
+/**
+ * A list of appointments that enforces uniqueness between its elements and does not allow nulls.
+ * A person is considered unique by comparing using {@code Appointment#equals(Appointment)}.
+ *
+ */
+public class UniqueAppointmentList implements Iterable<Appointment> {
+    private final ObservableList<Appointment> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Appointment> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Checks if the list contains the specified appointment.
+     *
+     * @param toCheck The appointment to check.
+     * @return True if the list contains the appointment, false otherwise.
+     * @throws NullPointerException if the appointment to check is null.
+     */
+    public boolean contains(Appointment toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    /**
+     * Adds an appointment to the list.
+     *
+     * @param toAdd The appointment to add.
+     * @throws NullPointerException if the appointment is null.
+     */
+    public void add(Appointment toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateAppointmentsException();
+        }
+        internalList.add(toAdd);
+
+    }
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedAppointment}.
```

Comment:
Should be replacing the appointment in the list with editedAppointment?

Reply:
oh right i copied over the method from person LMAO, will fix it now